baseapp/baseapp.go

```diff
@@ -810,6 +812,10 @@ func (app *BaseApp) endBlock(ctx context.Context) (sdk.EndBlock, error) {
 	return endblock, nil
 }
 
+type HasNestedMsgs interface {
+	GetMsgs() ([]sdk.Msg, error)
```

Comment:
Not sure whats the best place to define this 🤨

Reply:
it is a good practice to define the type where it is used. You can also consider making it private or inline it into the function where it is used. In this case there is the same interface in types called HasMsgs.

Reply:
hmm `HasMsgs` is pretty similar but doesn't return an error. I think it was designed with tx in mind. 
https://github.com/cosmos/cosmos-sdk/blob/83c4b9bfb42d9dbce91eb08a7799a99ac15b94cb/types/tx_msg.go#L48
Makes me wonder if we should update `HasMsgs` or stick with this 'duplicate'. If we opt for both interfaces, baseapp is not the ideal place for `HasNestedMsgs` as it may be useful for ante handlers.
@tac0turtle @julienrbrt

Reply:
what is the possible error that would be returned here? feel like it wouldnt return an error and just be length of 0

Reply:
Opted for that because current messages with nested messages within the SDK have defined that method:

https://github.com/cosmos/cosmos-sdk/blob/e8222c8092dedd4340e9599b38be414fac2293d8/x/gov/types/v1/proposal.go#L62-L65

https://github.com/cosmos/cosmos-sdk/blob/b536d1153202867cab457316155d7b68be57c04e/x/group/proposal.go#L10-L13

I guess we should standardise this and document it.